docs/dnspolicy/dnspolicy.md

```diff
@@ -0,0 +1,281 @@
+# Kuadrant DNSPolicy
+
+The DNSPolicy is a [GatewayAPI](https://gateway-api.sigs.k8s.io/) policy that uses `Direct Policy Attachment` as defined in the [policy attachment mechanism](https://gateway-api.sigs.k8s.io/v1alpha2/references/policy-attachment/) standard.
+This policy is used to provide dns management for gateway listeners by managing the lifecycle of dns records in external dns providers such as AWS Route53 and Google DNS.
+
+## How it works
+
+A DNSPolicy and its targeted Gateway API networking resource contain all the statements to configure both the ingress gateway and the external DNS service.
+
+### The DNSPolicy custom resource
+
+#### Overview
+
+The `DNSPolicy` spec includes the following parts:
+
+* A reference to an existing Gateway API resource (`spec.targetRef`)
+* DNS Routing Strategy (`spec.routingStrategy`)
+* LoadBalancing specification (`spec.loadBalancing`)
+* HealthCheck specification (`spec.healthCheck`)
+
+#### High-level example and field definition
+
+```yaml
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+metadata:
+  name: my-dns-policy
+spec:
+  # reference to an existing networking resource to attach the policy to
+  # it can only be a Gateway API Gateway resource
+  # it can only refer to objects in the same namespace as the DNSPolicy
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
+    name: mygateway
+   
+  # (optional) routing strategy to use when creating DNS records, defaults to `loadbalanced`
+  # determines what DNS records are created in the DNS provider
+  # check out Kuadrant RFC 0005 https://github.com/Kuadrant/architecture/blob/main/rfcs/0005-single-cluster-dnspolicy.md to learn more about the Routing Strategy field
+  # One-of: simple, loadbalanced.
+  routingStrategy: loadbalanced
```

Comment:
Not to do with this doc, but I wonder if part of v1alpha2 if we should move this default to be simple rather than load balanced?

Reply:
I think it would make sense that the simple routing strategy, even more so if the policies end up in a different repo from the OCM multi cluster stuff like we intend to do.